File: algorithm/binary-search/binary_search.py
def binary_search(list, target):
    low = 0
    high = len(list) - 1

    while low < high:
        mid = (high + low) // 2 # ‘//’无论是否整除，返回的都是int
        guess = list[mid]
        if target < guess:
            high = mid - 1
        if target == guess:
            return mid
        else:
            low = mid + 1
    return None

print(binary_search([1,2,3,4,5,6,7,9,10,11], 5))

File: algorithm/binary-search/binary-search.js
/**
 * 查找元素在有序数组中的序号
 */
function binarySearch(list, target) { 
    let low = 0,
        high = list.length - 1

    while (low <= high) {
        const mid = Math.floor((high + low) / 2)
        const midItem = list[mid]
        if (target < midItem) {
            high = mid - 1
        } else if (target === midItem) {
            return mid
        } else {
            low = mid + 1
        }
    }
    return null
}